ananse/influence.py

```diff
@@ -216,197 +323,366 @@ def filter_TF(scores_df, network=None, tpmfile=None, tpm=20, overlap=0.98):
     return scores_df
 
 
-def plot_influscore(infile, outfile):
-    """Plot TF influence score to expression."""
-
-    mogrify = pd.read_table(infile, index_col="factor")
-    mogrify = mogrify.dropna()
-    factors = list(mogrify.sort_values("sumScaled").tail(20).index)
-    # factors = list(mogrify.sort_values("sumScaled").tail(20).index)
-    xcol = "factor_fc"
-    plt.figure(figsize=(8, 6))
-    sns.regplot(
-        data=mogrify,
-        x=xcol,
-        y="sumScaled",
-        fit_reg=False,
-        scatter_kws={"s": mogrify["directTargets"] / 10, "alpha": 0.5},
-    )
-    x = mogrify.loc[factors, xcol]
-    y = mogrify.loc[factors, "sumScaled"]
-    texts = []
-    for s, xt, yt in zip(factors, x, y):
-        texts.append(plt.text(xt, yt, s))
-    adjust_text(texts, arrowprops=dict(arrowstyle="-", color="black"))
-    plt.xlabel("Log2 fold change of TF")
-    plt.ylabel("Influence score")
-    plt.savefig(outfile, dpi=300)
-
-
 class Influence(object):
     def __init__(
-        self, outfile, degenes, Gbf=None, Gaf=None, filter=False, edges=100000, ncore=1
+        self,
+        outfile,
+        degenes,
+        gene_gtf=None,
+        grn_source_file=None,
+        grn_target_file=None,
+        filter_tfs=False,  # TODO: variable not exposed in CLI
+        edges=100_000,
+        ncore=1,
+        sort_by="prob",
+        padj_cutoff=0.05,
+        full_output=False,
     ):
-
         self.ncore = ncore
-        logger.info(f"Reading network(s), using top {edges} edges.")
+        self.gene_gtf = gene_gtf
+        self.full_output = full_output
+        self.outfile = outfile
+        self.filter_tfs = filter_tfs
+
         # Load GRNs
-        if Gbf is None and Gaf is not None:
-            self.G = read_network(Gaf, edges=edges)
-            logger.warning("You only provide the target network!")
-        elif Gaf is None and Gbf is not None:
-            self.G = read_network(Gbf, edges=edges)
+        if grn_target_file is None and grn_source_file is None:
+            logger.error("You should provide at least one ANANSE network file!")
+            sys.exit(1)
+        logger.info(f"Loading network data, using the top {edges} edges")
+        if grn_source_file is None:
+            self.grn = read_network(grn_target_file, edges=edges)
+            logger.warning("You only provided the target network!")
+        elif grn_target_file is None:
+            self.grn = read_network(grn_source_file, edges=edges)
```

Comment:
This makes no sense. If only one network is provided, it should by default be the target network.